lib/ClangImporter/ClangImporter.cpp

```diff
@@ -3257,6 +3257,48 @@ void ClangImporter::Implementation::lookupAllObjCMembers(
   }
 }
 
+bool ClangImporter::Implementation::loadNamedMembers(
+    const Decl *D, DeclName N, uint64_t contextData,
+    TinyPtrVector<ValueDecl *> &Members) {
+
+  auto *DC = dyn_cast<DeclContext>(D);
+  if (!DC)
+    return true;
+
+  auto *CD = D->getClangDecl();
+  if (!CD)
+    return true;
```

Comment:
These are things you should assert.

Reply:
All 4 of them? DeclContext existing, ClangDecl existing, Submodule existing, and Lookuptable existing? I wasn't sure which of those are "sometimes not there" situations.

Reply:
Hm.

- A non-DeclContext can't have members.
- A non-Clang decl won't have the importer as its lazy loader.
- But a Clang decl from the bridging header won't have a Clang module (though it *will* have a lookup table).
- I *think* we have lookup tables for everything by now, so that actually ought to be safe to assert as well. We probably want to know if there isn't one.